packages/components/src/internal/components/domainproperties/SampleFieldOptions.tsx

```diff
@@ -50,16 +65,43 @@ export class SampleFieldOptions extends PureComponent<SampleFieldProps, State> {
                 .filter(st => st.type.isInteger()) // Remove rowId duplicates
                 .toList();
 
-            this.setState({ loadingState: LoadingState.LOADED, sampleTypes });
+            this.setState({
+                loadingState: LoadingState.LOADED,
+                sampleTypes,
+                validateLookup: field.isNew() || !!field.lookupValidator,
+            });
         } catch (e) {
             console.error('Failed to load sample field information', e);
             this.setState({ loadingState: LoadingState.LOADED });
         }
     };
 
+    onFieldChange = (evt: any): void => {
+        this.props.onChange?.(evt.target.id, evt.target.value);
```

Comment:
Looks like `onChange` is required and always supplied. Consider:

```ts
onFieldChange: React.ChangeEventHandler<HTMLSelectElement> = (evt): void => {
    this.props.onChange(evt.target.id, evt.target.value);
};
```

Reply:
Done.